framework/scripts/get_repo_revision.py

```diff
@@ -21,13 +21,25 @@ def shellCommand( command, cwd=None ):
 
         return p.communicate()[0]
 
+def hideSignature():
+    """
+    Conditionally returns the flag --no-show-signature if the version
+    of git is new enough to support that option.
+    """
+    try:
+        gitVersion = shellCommand( 'git --version' )
+        if LooseVersion(gitVersion) >= LooseVersion("2.9"):
+            return " --no-show-signature "
+    except:
+        return None
```

Comment:
shouldn't this return " " rather than None?
You can't really do a `"foo" + None + "bar"`
Also, move the last return statement outside of the except clause.

Reply:
The other methods return `None` on exception, probably so that things fail if there is a problem with git. But I agree that the if-statement should have the else-branch returning an empty string.

Reply:
Actually, it has to return `" "` (a single space), because the caller does not add spaces. Or the caller needs to take care of adding them when we return `""`.

Reply:
I looked into this. All of the other `None`s are ok. They are handled correctly. Mine wasn't so I changed it to an empty string.